From 4ce45b96e2296d135417b9506b0d22e7eed0497f Mon Sep 17 00:00:00 2001
From: Antonio Terceiro <terceiro@debian.org>
Date: Mon, 18 Jan 2021 21:44:59 -0300
Subject: [PATCH] Provide dummy plugin files for backwards compatibility

libjs-jquery-flot 0.8 provided the main file, jquery.flot.js, and one
had to include the plugin files explicitly. In version 4.2,
jquery.flot.js already includes the plugins, and existing code that
includes them again cause the charts to break in subtle ways.

Providing empty files in the place of the plugins allow existing usage
to continue working.

Thanks to W. Martin Borgert for the idea.

Closes: #980294
---
 debian/libjs-jquery-flot.install |  6 +-----
 debian/rules                     | 13 ++++++++++---
 2 files changed, 11 insertions(+), 8 deletions(-)

diff --git a/debian/libjs-jquery-flot.install b/debian/libjs-jquery-flot.install
index 102d532..3eaf78d 100644
--- a/debian/libjs-jquery-flot.install
+++ b/debian/libjs-jquery-flot.install
@@ -1,7 +1,3 @@
-source/jquery.canvaswrapper*.js usr/share/javascript/jquery-flot/
-source/jquery.colorhelpers*.js usr/share/javascript/jquery-flot/
-source/jquery.flot.*.js usr/share/javascript/jquery-flot/
-source/*.map usr/share/javascript/jquery-flot/
 dist/es5/*.js usr/share/javascript/jquery-flot/
-dist/es5/*.js.map usr/share/javascript/jquery-flot/
 dist/es5/*.map usr/share/javascript/jquery-flot/
+plugins/*.js   usr/share/javascript/jquery-flot/
diff --git a/debian/rules b/debian/rules
index 4bc2f33..18e244e 100755
--- a/debian/rules
+++ b/debian/rules
@@ -5,15 +5,22 @@
 
 override_dh_auto_build:
 	gulp build
-	cd source && for f in *.js; do \
-		echo "Minify source/$$f"; \
-		uglifyjs --source-map -o $${f%%.js}.min.js $$f; \
+	# Provide empty files for plugins for backwards compatibility since
+	# they are now included with the main jquery.flot.js file
+	mkdir plugins
+	cd source && for f in jquery.flot.*.js; do \
+		echo "Provide dummy file for $$f" ; \
+		touch ../plugins/$${f%%.js}.min.js ../plugins/$$f; \
 	done
 	cd dist/es5 && for f in *.js; do \
 		echo "Minify dist/es5/$$f"; \
 		uglifyjs --source-map -o $${f%%.js}.min.js $$f; \
 	done
 
+override_dh_auto_clean:
+	dh_auto_clean
+	$(RM) -r plugins
+
 get-orig-source:
 	OUTDIR=$$PWD ; \
 	MAKEFILE=`echo $(MAKEFILE_LIST) | awk '{ print $$1 }'` ; \
-- 
2.30.0

